optional/src/test/java/org/dmfs/optional/composite/ZippedTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.optional.composite;
+
+import org.dmfs.jems.function.BiFunction;
+import org.dmfs.optional.Absent;
+import org.dmfs.optional.Present;
+import org.junit.Test;
+
+import java.util.NoSuchElementException;
+
+import static org.dmfs.jems.mockito.doubles.TestDoubles.dummy;
+import static org.dmfs.optional.Absent.absent;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+
+/**
+ * @author Marten Gajda
+ */
+public class ZippedTest
+{
+    @Test
+    public void testIsPresent() throws Exception
+    {
+        assertThat(new Zipped<>(new Present<>("val1"), new Present<>("val2"), dummy(BiFunction.class)).isPresent(), is(true));
+        assertThat(new Zipped<>(absent(), new Present<>("val2"), dummy(BiFunction.class)).isPresent(), is(false));
+        assertThat(new Zipped<>(new Present<>("val1"), absent(), dummy(BiFunction.class)).isPresent(), is(false));
+        assertThat(new Zipped<>(absent(), absent(), dummy(BiFunction.class)).isPresent(), is(false));
+    }
+
+
+    @Test
+    public void testValue() throws Exception
+    {
+        assertThat(new Zipped<>(new Present<>("val1"), new Present<>("val2"), stringBiFunction("val1", "val2", "result")).value("fallback"), is("result"));
+        assertThat(new Zipped<>(Absent.<String>absent(), new Present<>("val2"), stringBiFunction("val1", "val2", "result")).value("fallback"), is("fallback"));
+        assertThat(new Zipped<>(new Present<>("val1"), Absent.<String>absent(), stringBiFunction("val1", "val2", "result")).value("fallback"), is("fallback"));
+        assertThat(new Zipped<>(Absent.<String>absent(), Absent.<String>absent(), stringBiFunction("val1", "val2", "result")).value("fallback"),
+                is("fallback"));
+    }
+
+
+    @Test
+    public void testValue1() throws Exception
+    {
+        assertThat(new Zipped<>(new Present<>("val1"), new Present<>("val2"), stringBiFunction("val1", "val2", "result")).value(), is("result"));
```

Comment:
This test case is in the previous test method as well.

Reply:
Not really, the previous test uses `value(default)` this one doesn't. 
But why didn't I use `OptionalMatcher` in the first place !?!? I'll update that.

Reply:
Ah, ok, sorry, I didn't notice that.

Reply:
I've decided that the test is fine that way.